packages/vitest/src/node/config/resolveConfig.ts

```diff
@@ -238,10 +238,23 @@ export function resolveConfig(
     }
   }
 
+  // apply browser CLI options only if the config already has the browser config and not disabled manually
+  if (
+    vitest._cliOptions.browser
+    && resolved.browser
+    // if enabled is set to `false`, but CLI overrides it, then always override it
+    && (resolved.browser.enabled !== false || vitest._cliOptions.browser.enabled)
+  ) {
+    resolved.browser = mergeConfig(
+      resolved.browser,
+      vitest._cliOptions.browser,
+    ) as ResolvedBrowserOptions
+  }
+
+  resolved.browser ??= {} as any
```

Comment:
btw this check is twice in this file, has been for a while:

https://github.com/vitest-dev/vitest/blob/d5f68a961ab55fd57d61f2125679e7733ea53e97/packages/vitest/src/node/config/resolveConfig.ts#L822